fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/KeySpaceDirectory.java

```diff
@@ -719,6 +720,12 @@ protected static boolean areEqual(Object o1, Object o2) {
             }
         }
 
+        // Handle ANY_VALUE specially - typeOf does not support ANY_VALUE
+        boolean isAnyValue = (o1 == ANY_VALUE || o2 == ANY_VALUE);
+        if (isAnyValue) {
+            return Objects.equals(o1, o2);
```

Comment:
As a defensive coding we can add `equals` and `hashCode` to `AnyValue`, though all reasonable implementations should use the static constant...